From fe61ca28d8711d3f2ed2cb7ea5d127dffee034df Mon Sep 17 00:00:00 2001
From: Jean-Yves Avenard <jyavenard@mythtv.org>
Date: Sat, 28 Jun 2014 15:32:42 +1000
Subject: [PATCH 13/16] Properly handle when libmythbluray is unable to open a
 bluray, should libaacs or libbd+ failed to decrypt.

It would otherwise leave the BD drive in an unmounted state, and sometimes a crash when attempting to read from the drive
---
 mythtv/libs/libmythtv/Bluray/bdringbuffer.cpp |   12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/mythtv/libs/libmythtv/Bluray/bdringbuffer.cpp b/mythtv/libs/libmythtv/Bluray/bdringbuffer.cpp
index 25c9bb5..14c4b00 100644
--- a/mythtv/libs/libmythtv/Bluray/bdringbuffer.cpp
+++ b/mythtv/libs/libmythtv/Bluray/bdringbuffer.cpp
@@ -322,6 +322,18 @@ bool BDRingBuffer::OpenFile(const QString &lfilename, uint retry_ms)
     m_topMenuSupported   = false;
     m_firstPlaySupported = false;
     const BLURAY_DISC_INFO *discinfo = bd_get_disc_info(bdnav);
+    if (!discinfo || (discinfo->aacs_detected && !discinfo->aacs_handled) ||
+        (discinfo->bdplus_detected && !discinfo->bdplus_handled))
+    {
+        // couldn't decrypt bluray
+        bd_close(bdnav);
+        bdnav = NULL;
+        lastError = tr("Could not open Blu-ray device %1, failed to decrypt")
+                    .arg(filename);
+        rwlock.unlock();
+        mythfile_open_register_callback(filename.toLocal8Bit().data(), this, NULL);
+        return false;
+    }
 
     // The following settings affect HDMV navigation
     // (default audio track selection,
-- 
1.7.10.2

